src/main/io/ledstrip.c

```diff
@@ -257,622 +228,663 @@ void pgResetFn_ledConfigs(ledConfig_t *instance)
+            if (feature(FEATURE_FAILSAFE) && failsafeIsActive())
+                warningFlags |= 1 << WARNING_FAILSAFE;
+            if (!ARMING_FLAG(ARMED) && !ARMING_FLAG(OK_TO_ARM))
+                warningFlags |= 1 << WARNING_ARMING_DISABLED;
         }
+        *timer += LED_STRIP_HZ(10);
     }
 
-    if (warningFlags || warningFlashCounter > 0) {
-        const hsvColor_t *warningColor = &hsv_black;
+    if (warningFlags) {
+        const hsvColor_t *warningColor =  &HSV(BLACK);
 
-        if ((warningFlashCounter & 1) == 0) {
-            if (warningFlashCounter < 4 && (warningFlags & WARNING_FLAG_ARMING_DISABLED)) {
-                warningColor = &hsv_green;
-            }
-            if (warningFlashCounter >= 4 && warningFlashCounter < 12 && (warningFlags & WARNING_FLAG_LOW_BATTERY)) {
-                warningColor = &hsv_red;
-            }
-            if (warningFlashCounter >= 12 && warningFlashCounter < 16 && (warningFlags & WARNING_FLAG_FAILSAFE)) {
-                warningColor = &hsv_yellow;
-            }
-        }  else {
-            if (warningFlashCounter >= 12 && warningFlashCounter < 16 && (warningFlags & WARNING_FLAG_FAILSAFE)) {
-                warningColor = &hsv_blue;
+        bool colorOn = (warningFlashCounter % 2) == 0;   // w_w_
+        warningFlags_e warningId = warningFlashCounter / 4;
+        if(warningFlags & (1 << warningId)) {
+            switch(warningId) {
+                case WARNING_ARMING_DISABLED:
+                    warningColor = colorOn ? &HSV(GREEN)  : &HSV(BLACK);
+                    break;
+                case WARNING_LOW_BATTERY:
+                    warningColor = colorOn ? &HSV(RED)    : &HSV(BLACK);
+                    break;
+                case WARNING_FAILSAFE:
+                    warningColor = colorOn ? &HSV(YELLOW) : &HSV(BLUE);
+                    break;
+                default:;
             }
         }
 
-        for (int ledIndex = 0; ledIndex < ledCount; ledIndex++) {
-            const ledConfig_t *ledConfig = ledConfigs(ledIndex);
-            if (!(ledConfig->flags & LED_FUNCTION_WARNING))
-                continue;
-
-            setLedHsv(ledIndex, warningColor);
-        }
-    }
-
-    if (updateNow && (warningFlags || warningFlashCounter)) {
-        warningFlashCounter++;
-        if (warningFlashCounter == 20) {
-            warningFlashCounter = 0;
-        }
+        applyLedHsv(LED_FUNCTION_WARNING, warningColor);
     }
 }
 
-void applyLedGpsLayer(bool updateNow)
+static void applyLedBatteryLayer(bool updateNow, uint32_t *timer)
 {
-#ifdef CC3D
-    return;
-#endif
-    static uint8_t gpsFlashCounter = 0;
-    static uint8_t gpsPauseCounter = 0;
-    const uint8_t blinkPauseLength = 4;
+    static bool batteryFlash = false;
+    static uint8_t battState;
 
-    const hsvColor_t *gpsColor = colors(specialColors(0)->background);
+    if (updateNow) {
+        battState = getBatteryState();
+        if (battState == BATTERY_OK) {
```

Comment:
Using `switch((batteryState_e )battState)` here may be easier to read, even when it means duplicating `batteryFlash = !batteryFlash;`